src/controllers/users.go

```diff
@@ -19,13 +21,32 @@ func NewUsersController(db *gorm.DB) *UsersController {
 
 // Create new user
 func (c *UsersController) Create(email string, password string) error {
+	// Check if email is already registered
+	existingUser := &models.User{}
+	result := c.db.Where("email = ?", email).First(existingUser)
```

Comment:
extract into a separate findUser function because we'll use that in other places too

Reply:
Fair enough.